bridge-circuit-host/src/docker.rs

```diff
@@ -221,7 +235,402 @@ const ID_BN254_FR_BITS: [&str; 254] = [
+        ));
+    }
+    let remote_config_digest_json = String::from_utf8(remote_config_digest_output.stdout)
+        .wrap_err("Failed to parse remote image manifest as UTF-8")?;
+    let remote_config_digest: String =
+        serde_json::from_str::<serde_json::Value>(&remote_config_digest_json)?
+            .get("config")
+            .and_then(|c| c.get("digest"))
+            .and_then(|d| d.as_str())
+            .ok_or_else(|| eyre!("Failed to get remote config digest"))?
+            .to_string();
+
+    // 2. Inspect local tar digest if it exists
+    let mut need_pull = true;
+    if tar_file_path.exists() {
+        let local_config_digest_output = Command::new("skopeo")
+            .arg("inspect")
+            .arg("--raw")
+            .arg(format!(
+                "docker-archive:{}",
+                tar_file_path.to_string_lossy()
+            ))
+            .output()
+            .wrap_err("skopeo inspect could not be executed")?;
+        if local_config_digest_output.status.success() {
+            let local_config_digest_json = String::from_utf8(local_config_digest_output.stdout)
+                .wrap_err("Failed to parse local image manifest as UTF-8")?;
+            let local_config_digest: String =
+                serde_json::from_str::<serde_json::Value>(&local_config_digest_json)?
+                    .get("config")
+                    .and_then(|c| c.get("digest"))
+                    .and_then(|d| d.as_str())
+                    .ok_or_else(|| eyre!("Failed to get local config digest"))?
+                    .to_string();
+
+            if local_config_digest == remote_config_digest {
+                tracing::info!("Local tar config digest matches remote, skipping pull");
+                need_pull = false;
+            } else {
+                tracing::warn!(
+                    "Local tar config digest mismatch, deleting and re-pulling: local={}, remote={}",
+                    local_config_digest,
+                    remote_config_digest
+                );
+                fs::remove_file(&tar_file_path).wrap_err("Failed to delete local tar")?;
+            }
+        } else {
+            tracing::warn!("Failed to inspect local tar config digest, re-pulling");
+            fs::remove_file(&tar_file_path).ok();
+        }
+    }
+
+    // 3. Pull with skopeo if needed
+    if need_pull {
+        tracing::info!("Pulling image via skopeo...");
+        let pull_output = Command::new("skopeo")
+            .arg("copy")
+            .arg(format!("docker://{image_digest}"))
+            .arg(format!(
+                "docker-archive:{}",
+                tar_file_path.to_string_lossy()
+            ))
+            .output()
+            .wrap_err("skopeo copy could not be executed")?;
+        if !pull_output.status.success() {
+            return Err(eyre!("skopeo copy failed: {:?}", pull_output));
+        }
+    }
+
+    let modified_tar_path = remove_symlinks_from_image_tar(&tar_file_path)?;
```

Comment:
I think we should extract this to:

`async fn pull_or_load_image(image_digest: String) -> PathBuf` 

Then, for the Dockerfile.automation, we should embed the properly formatted tar files. Can we add a CLI command for cache_prover_images that'll run `pull_or_load_image` for all prover images, and cache them in the home directory.